<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from robot_version1.xacro           | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="robot_version1_proyect_grip">

<link name="world"/>

<joint name="world_joint" type="fixed">
  <parent link="world"/>
  <child link="dummy_base"/>
  <origin rpy="1.5 0 0" xyz="0 0 -0.1"/>
</joint>

  <material name="silver">
    <color rgba="0.700 0.700 0.700 1.000"/>
  </material>
  <transmission name="motor1_tran">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="motor1">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor1_actr">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="motor2_tran">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="motor2">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor2_actr">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="motor3_tran">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="motor3">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor3_actr">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="motor4_tran">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="motor4">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor4_actr">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="motor5_tran">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="motor5">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor5_actr">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="motor6_tran">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="motor6">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor6_actr">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <gazebo>
    <plugin filename="libgazebo_ros_control.so" name="control"/>
  </gazebo>

  
  <gazebo reference="base_link">
    <material>Gazebo/Silver</material>
    <mu1>0.2</mu1>
    <mu2>0.2</mu2>
    <self_collide>true</self_collide>
    <gravity>true</gravity>
  </gazebo>
  <gazebo reference="link1_1">
    <material>Gazebo/Silver</material>
    <mu1>0.2</mu1>
    <mu2>0.2</mu2>
    <self_collide>true</self_collide>
  </gazebo>
  <gazebo reference="link2_1">
    <material>Gazebo/Silver</material>
    <mu1>0.2</mu1>
    <mu2>0.2</mu2>
    <self_collide>true</self_collide>
  </gazebo>
  <gazebo reference="link3_1">
    <material>Gazebo/Silver</material>
    <mu1>0.2</mu1>
    <mu2>0.2</mu2>
    <self_collide>true</self_collide>
  </gazebo>
  <gazebo reference="link4_1">
    <material>Gazebo/Silver</material>
    <mu1>0.2</mu1>
    <mu2>0.2</mu2>
    <self_collide>true</self_collide>
  </gazebo>
  <gazebo reference="link5_1">
    <material>Gazebo/Silver</material>
    <mu1>0.2</mu1>
    <mu2>0.2</mu2>
    <self_collide>true</self_collide>
  </gazebo>
  <gazebo reference="link6_1">
    <material>Gazebo/Silver</material>
    <mu1>0.2</mu1>
    <mu2>0.2</mu2>
    <self_collide>true</self_collide>
  </gazebo>

<link name="dummy_base">
  </link>

  <gazebo reference="dummy_base">
  <static>true</static>
</gazebo>

  <link name="base_link">
    <inertial>
      <origin rpy="0 0 0" xyz="-1.1580046200244058e-17 0.024999999999999994 5.790023100122029e-18"/>
      <mass value="3.082687791334984"/>
      <inertia ixx="0.002569" ixy="-0.0" ixz="-0.0" iyy="0.003853" iyz="-0.0" izz="0.002569"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="file:///home/user/lab_ws/src/frlabs/lab7/meshes/base_link.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="silver"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="file:///home/user/lab_ws/src/frlabs/lab7/meshes/base_link.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <link name="link1_1">
    <inertial>
      <origin rpy="0 0 0" xyz="9.008917649016016e-18 0.0437607595854386 0.035413080138187114"/>
      <mass value="7.924962450619459"/>
      <inertia ixx="0.016525" ixy="0.0" ixz="0.0" iyy="0.016428" iyz="-0.002169" izz="0.010003"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="-0.0 -0.05 -0.0"/>
      <geometry>
        <mesh filename="file:///home/user/lab_ws/src/frlabs/lab7/meshes/link1_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="silver"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="-0.0 -0.05 -0.0"/>
      <geometry>
        <mesh filename="file:///home/user/lab_ws/src/frlabs/lab7/meshes/link1_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <link name="link2_1">
    <inertial>
      <origin rpy="0 0 0" xyz="-8.314822887740735e-18 0.016342334380787568 0.12980540062847232"/>
      <mass value="17.173025824624403"/>
      <inertia ixx="0.10519" ixy="-0.0" ixz="0.0" iyy="0.093316" iyz="-0.020118" izz="0.03334"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0.0 -0.1 -0.1"/>
      <geometry>
        <mesh filename="file:///home/user/lab_ws/src/frlabs/lab7/meshes/link2_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="silver"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0.0 -0.1 -0.1"/>
      <geometry>
        <mesh filename="file:///home/user/lab_ws/src/frlabs/lab7/meshes/link2_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <link name="link3_1">
    <inertial>
      <origin rpy="0 0 0" xyz="1.013394198499686e-17 0.1027347532259798 0.01991771451421981"/>
      <mass value="14.090338033289418"/>
      <inertia ixx="0.060349" ixy="-0.0" ixz="0.0" iyy="0.028484" iyz="-0.013683" izz="0.049478"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="-0.0 -0.2 -0.3"/>
      <geometry>
        <mesh filename="file:///home/user/lab_ws/src/frlabs/lab7/meshes/link3_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="silver"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="-0.0 -0.2 -0.3"/>
      <geometry>
        <mesh filename="file:///home/user/lab_ws/src/frlabs/lab7/meshes/link3_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <link name="link4_1">
    <inertial>
      <origin rpy="0 0 0" xyz="0.0 0.016342334380787638 0.12980540062847234"/>
      <mass value="17.173025824624403"/>
      <inertia ixx="0.10519" ixy="0.0" ixz="0.0" iyy="0.093316" iyz="-0.020118" izz="0.03334"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0.0 -0.35 -0.4"/>
      <geometry>
        <mesh filename="file:///home/user/lab_ws/src/frlabs/lab7/meshes/link4_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="silver"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0.0 -0.35 -0.4"/>
      <geometry>
        <mesh filename="file:///home/user/lab_ws/src/frlabs/lab7/meshes/link4_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <link name="link5_1">
    <inertial>
      <origin rpy="0 0 0" xyz="0.0 0.11444820806225936 0.003509195158339584"/>
      <mass value="14.090338033289422"/>
      <inertia ixx="0.071182" ixy="-0.0" ixz="0.0" iyy="0.017844" iyz="-0.004648" izz="0.07095"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="-0.0 -0.45 -0.6"/>
      <geometry>
        <mesh filename="file:///home/user/lab_ws/src/frlabs/lab7/meshes/link5_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="silver"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="-0.0 -0.45 -0.6"/>
      <geometry>
        <mesh filename="file:///home/user/lab_ws/src/frlabs/lab7/meshes/link5_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <link name="link6_1">
    <inertial>
      <origin rpy="0 0 0" xyz="1.7953246794872857e-11 3.3854696823709673e-10 0.02603270456929907"/>
      <mass value="3.1925877913349834"/>
      <inertia ixx="0.002669" ixy="0.0" ixz="-0.0" iyy="0.00271" iyz="-0.0" izz="0.003902"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0.0 -0.65 -0.65"/>
      <geometry>
        <mesh filename="file:///home/user/lab_ws/src/frlabs/lab7/meshes/link6_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="silver"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0.0 -0.65 -0.65"/>
      <geometry>
        <mesh filename="file:///home/user/lab_ws/src/frlabs/lab7/meshes/link6_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>

  <joint name="dummy_joint" type="fixed">
     <parent link="dummy_base"/>
     <child link="base_link"/>
  </joint>

  <joint name="motor1" type="revolute">
    <origin rpy="0 0 0" xyz="0.0 0.05 0.0"/>
    <parent link="base_link"/>
    <child link="link1_1"/>
    <axis xyz="0.0 1.0 0.0"/>
    <limit lower="-3.14" upper="3.14" effort="5.0" velocity="2.0"/>
  </joint>
  <joint name="motor2" type="continuous">
    <origin rpy="0 0 0" xyz="-0.0 0.05 0.1"/>
    <parent link="link1_1"/>
    <child link="link2_1"/>
    <axis xyz="0.0 0.0 1.0"/>
  </joint>
  <joint name="motor3" type="continuous">
    <origin rpy="0 0 0" xyz="0.0 0.1 0.2"/>
    <parent link="link2_1"/>
    <child link="link3_1"/>
    <axis xyz="0.0 1.0 0.0"/>
  </joint>
  <joint name="motor4" type="continuous">
    <origin rpy="0 0 0" xyz="-0.0 0.15 0.1"/>
    <parent link="link3_1"/>
    <child link="link4_1"/>
    <axis xyz="0.0 0.0 1.0"/>
  </joint>
  <joint name="motor5" type="continuous">
    <origin rpy="0 0 0" xyz="0.0 0.1 0.2"/>
    <parent link="link4_1"/>
    <child link="link5_1"/>
    <axis xyz="0.0 1.0 0.0"/>
  </joint>
  <joint name="motor6" type="continuous">
    <origin rpy="0 0 0" xyz="-0.0 0.2 0.05"/>
    <parent link="link5_1"/>
    <child link="link6_1"/>
    <axis xyz="0.0 0.0 1.0"/>
  </joint>




</robot>

